File: frontend/lib/core/utils/mime_type.dart
/// Best-effort mimetype guess from a file extension — used wherever a
/// picked file's bytes need a Content-Type for multipart upload (file_picker
/// v12's PlatformFile no longer exposes mimetype directly, only a name).
String? extensionOf(String fileName) {
  final dot = fileName.lastIndexOf('.');
  if (dot < 0 || dot == fileName.length - 1) return null;
  return fileName.substring(dot + 1);
}

String mimeTypeForExtension(String? ext) {
  switch ((ext ?? '').toLowerCase()) {
    case 'pdf':
      return 'application/pdf';
    case 'jpg':
    case 'jpeg':
      return 'image/jpeg';
    case 'png':
      return 'image/png';
    case 'tif':
    case 'tiff':
      return 'image/tiff';
    case 'docx':
      return 'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
    case 'xlsx':
      return 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
    case 'xls':
      return 'application/vnd.ms-excel';
    case 'txt':
      return 'text/plain';
    default:
      return 'application/octet-stream';
  }
}

File: backend/utils/mimeType.js
/**
 * utils/mimeType.js
 * Filesystem/FTP/email intake sources have no browser-supplied
 * Content-Type header (unlike multipart uploads), so infer one from the
 * file extension — covers exactly the types services/ocr.service.js knows
 * how to extract text from, plus a safe binary-octet-stream fallback.
 */
const EXTENSION_MIME_MAP = {
  pdf: 'application/pdf',
  docx: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
  xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
  xls: 'application/vnd.ms-excel',
  txt: 'text/plain',
  png: 'image/png',
  jpg: 'image/jpeg',
  jpeg: 'image/jpeg',
  tif: 'image/tiff',
  tiff: 'image/tiff',
};

function mimeFromExtension(fileName) {
  const ext = (fileName.split('.').pop() || '').toLowerCase();
  return EXTENSION_MIME_MAP[ext] || 'application/octet-stream';
}

module.exports = { mimeFromExtension };
